src/custom.js

```diff
@@ -0,0 +1,272 @@
+ * * const {xpub, rootFingerprint, bip32Path} = interaction.parse({xpub: xpub...});
+ * console.log(xpub);
+ * // "xpub..."
+ * console.log(rootFingerprint);
+ * // "096aed5e"
+ * console.log(bip32Path);
+ * // "m/45'/0'/0'"
+ */
+export class CustomExportExtendedPublicKey extends CustomInteraction {
+  /**
+   * @param {object} options - options argument
+   * @param {string} options.network - bitcoin network (needed for derivations)
+   * @param {string} options.bip32Path - bip32Path to interrogate
+   */
+  constructor({ network, bip32Path }) {
+    super();
+    if ([MAINNET, TESTNET].find((net) => net === network)) {
+      this.network = network;
+    } else {
+      throw new Error("Unknown network.");
+    }
+    this.validationErrorMessages = [];
+    this.bip32Path = bip32Path;
+    const bip32PathError = validateBIP32Path(bip32Path);
+    if (bip32PathError.length) {
+      this.validationErrorMessages.push({
+        code: "custom.bip32_path.path_error",
+        text: bip32PathError,
+      });
+    }
+  }
+
+  isSupported() {
+    return this.validationErrorMessages.length === 0;
+  }
+
+  messages() {
+    const messages = super.messages();
+
+    if (this.validationErrorMessages.length) {
+      this.validationErrorMessages.map((e) => {
+        messages.push({
+          state: PENDING,
+          level: ERROR,
+          code: e.code,
+          text: e.text,
+        });
+      });
+    }
+
+    messages.push({
+      state: PENDING,
+      level: INFO,
+      code: "custom.import_xpub",
+      text: "Type or paste the extended public key here.",
+    });
+    return messages;
+  }
+
+  /**
+   * Parse the provided JSON and do some basic error checking
+   *
+   * @param {Object} data - JSON object with incoming data to be parsed (read: reformatted)
+   * @returns {Object} Object - ExtendedPublicKeyDerivation {xpub, bip32path, rootFingerprint}
+   */
+  parse(data) {
+    // build ExtendedPublicKey struct (validation happens in constructor)
+    let xpubClass;
+    let rootFingerprint;
+    try {
+      xpubClass = ExtendedPublicKey.fromBase58(data.xpub);
+    } catch (e) {
+      throw new Error("Not a valid ExtendedPublicKey.");
+    }
+    try {
+      const pkLen = xpubClass.pubkey.length;
+      // If no fingerprint is provided, we will assign one deterministically
+      const fakeRootFingerprint = xpubClass.pubkey.substring(pkLen - 8);
+
+      if (data.rootFingerprint === "" || data.rootFingerprint === undefined) {
+        rootFingerprint = fakeRootFingerprint;
+      } else {
+        validateRootFingerprint(data.rootFingerprint);
+        rootFingerprint = data.rootFingerprint;
+      }
+    } catch (e) {
+      throw new Error(
+        `Root fingerprint validation error: ${e.message.toLowerCase()}.`
+      );
+    }
+    const numSlashes = this.bip32Path.split("/").length;
+    const bipDepth = this.bip32Path.startsWith("m/")
+      ? numSlashes - 1
+      : numSlashes;
+
+    if (xpubClass.depth !== bipDepth) {
```

Comment:
Yes in basically the same way as any other interaction. 
The bip32path can't be determined from the xpub itself, it only has depth embedded inside. 

This is just a check that what the caller *said* the bip32path was **at least** matches the depth of what you provided because that's as much as we can check.

E.g. if you said m/45' and gave me m/45'/0'/0' -- we should say `hold on a minute...`

Reply:
> The bip32path can't be determined from the xpub itself, it only has depth embedded inside.

Oh yeah, duh. Sorry.